### PR TITLE
misc: Add checkout-pr.sh script for easier checkout

### PR DESCRIPTION
**Pull request by honggyukim**
It takes some time to get the patches for review but it'd be useful to have a script for that.

The script accepts only a parameter to get the pull request ID. It fetches from the source repo and checkouts to the fetched head.

### CODE REVIEW

File: misc/checkout-pr.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+if [ ! -x "$(command -v curl)" ]; then
+	echo "You need curl to run this script."
+	exit 1
+fi
+
+pr=$1
+pr_json="pr.json"
+
+curl -o $pr_json https://api.github.com/repos/namhyung/uftrace/pulls/$pr
+repo=$(jq -r '.head.repo.html_url' $pr_json)
```
Comment (namhyung) [inline misc/checkout-pr.sh:17]: What is `jq`?

**Review by namhyung — COMMENTED** [review]
Don't we have `misc/wget-pr.sh` already?

File: misc/checkout-pr.sh (hunk 1 above)
Comment (honggyukim) [reply misc/checkout-pr.sh:17]: https://jqlang.github.io/jq

**Review by honggyukim — COMMENTED** [review]
> Don't we have `misc/wget-pr.sh` already?

`wget-pr.sh` just downloads `.patch` file, but this fetches the remote branch then checkout to it. If the PR is a bit outdated then the downloaded `.patch` file can make conflicts.

Rather than this, we can just put the PR number then it will create a branch `pull/<pr>`, then checkout to it and we can easily see if the branch needs rebase or not.

The usage to checkout to #1756 is as follows.
```
$ ./misc/checkout-pr.sh 1756
```

**Comment by honggyukim** [comment]
This might be not needed if we're working on already registered remote repos but I felt it's tedious to fetch and checkout some new contributors' branch. This script makes the workflow much easier.

File: misc/checkout-pr.sh (hunk 1 above)
Comment (honggyukim) [reply misc/checkout-pr.sh:17]: You can easily understand by running the following command. It's to get the PR sender's uftrace repo.
```
$ curl -s https://api.github.com/repos/namhyung/uftrace/pulls/1947 | jq -r '.head.repo.html_url'
https://github.com/honggyukim/uftrace
```

Comment (honggyukim) [reply misc/checkout-pr.sh:17]: We can also get the branch name as follows.
```
$ curl -s https://api.github.com/repos/namhyung/uftrace/pulls/1947 | jq -r '.head.ref'
check/checkout-pr.sh
```

Comment (namhyung) [reply misc/checkout-pr.sh:17]: That's cool.  Is it ok to assume if it's installed on all modern distros?

**Review by namhyung — COMMENTED** [review]
Ok, I like you used `git checkout -B` to create or update the branch.

File: misc/checkout-pr.sh (hunk 1 above)
Comment (honggyukim) [reply misc/checkout-pr.sh:17]: Sorry for the late answer but I guess yes.

**Comment by honggyukim** [comment]
@namhyung I've just update it so please have a look again. Thanks.

**Review by namhyung — APPROVED** [review]
LGTM

**Pull request merged**